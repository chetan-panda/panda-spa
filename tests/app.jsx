jest.mock('socket.io-client', () => {
  return jest.fn(() => {
    return {
      // eslint-disable-next-line no-empty-function
      on: jest.fn(() => {}),
    }
  })
})

import App from '../lib/app'
import React from 'react'
import renderer from 'react-test-renderer'

test('should check default app structure', () => {
  const app = renderer.create(
    <App
      apiVersion='v0'
      isProduction={false}
      socketUrl='http://api.staging.chetanpanda.com'
    />
  )

  const tree = app.toJSON()
  expect(tree).toMatchSnapshot()
})
