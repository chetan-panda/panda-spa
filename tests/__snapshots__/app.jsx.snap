// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check default app structure 1`] = `
<div>
  <div
    id="content-manager"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "minHeight": "100vh",
      }
    }
  >
    <div
      id="page-about"
    >
      <div
        className="ui basic segment navigation"
        style={
          Object {
            "backgroundImage": "url(undefined)",
            "backgroundPosition": "center",
            "backgroundSize": "cover",
            "height": null,
            "left": 0,
            "marginBottom": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          className="navigation-overlay"
        />
        <div
          className="ui container navigation-container"
        >
          <div
            className="ui huge borderless inverted secondary menu"
          >
            <div
              className="item"
              onClick={[Function]}
            >
              <img
                alt={undefined}
                className="ui mini image"
                height={undefined}
                onClick={[Function]}
                src={null}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
                width={undefined}
              />
            </div>
            <div
              className="ui grid computer only"
            >
              <a
                className="link item"
                onClick={[Function]}
              >
                About
              </a>
              <a
                className="link item"
                onClick={[Function]}
              >
                Issues
              </a>
            </div>
            <div
              className="right menu"
            >
              <div
                className="ui grid computer only"
              >
                <div
                  className="item"
                  onClick={[Function]}
                >
                  <a
                    className="ui blue compact button"
                    href="http://eepurl.com/cVtGir"
                    onClick={[Function]}
                    tabIndex={undefined}
                  >
                    Join Us
                  </a>
                </div>
                <div
                  className="item"
                  onClick={[Function]}
                >
                  <a
                    className="ui orange compact button"
                    href="https://secure.actblue.com/donate/panda2018"
                    onClick={[Function]}
                    tabIndex={undefined}
                  >
                    Donate
                  </a>
                </div>
              </div>
              <div
                className="ui grid tablet mobile only"
              >
                <div
                  aria-busy={undefined}
                  aria-disabled={undefined}
                  aria-expanded={false}
                  aria-multiselectable={undefined}
                  className="ui simple dropdown-dark dropdown"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <div
                    className="text"
                  >
                     
                  </div>
                  <i
                    aria-hidden="true"
                    className="content icon"
                  />
                  <div
                    className="menu transition"
                  >
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={true}
                      className="selected item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        About
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Issues
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Join Us
                      </span>
                    </div>
                    <div
                      aria-checked={false}
                      aria-disabled={undefined}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Donate
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="page"
        id="page-home"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <div
          className="ui hidden divider"
        />
        <div
          className="ui hidden divider"
        />
      </div>
      <div
        className="ui basic inverted segment"
        style={
          Object {
            "backgroundColor": "#003666",
            "marginTop": 0,
            "paddingBottom": "3em",
            "paddingTop": "3em",
          }
        }
      >
        <div
          className="ui text container"
        >
          <h1
            className="ui inverted header"
          >
            Why I'm Running
          </h1>
          <div />
        </div>
      </div>
      <div
        className="ui basic segment"
        style={
          Object {
            "marginTop": 0,
            "paddingBottom": "3em",
            "paddingTop": "3em",
          }
        }
      >
        <div
          className="ui text container"
        >
          <h1
            className="ui header"
          >
            About Me
          </h1>
          <div />
        </div>
      </div>
    </div>
    <div
      className="ui basic inverted segment"
      style={
        Object {
          "backgroundColor": "#003666",
          "paddingBottom": "1.5em",
          "paddingTop": "1.5em",
        }
      }
    >
      <div
        className="ui center aligned container"
      >
        <p>
          <b>
            Paid for by the Committee to Elect Chetan Panda
          </b>
          <br />
          <a
            href="mailto:info@panda2018.com"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            <i
              aria-hidden="true"
              className="mail icon"
            />
            info@panda2018.com
          </a>
        </p>
        <p>
          If you'd  like to send a contribution by mail, please address it to:
          <br />
          Committee to Elect Chetan Panda, PO Box 29416 Austin, Texas 78755
        </p>
      </div>
    </div>
  </div>
</div>
`;
